packages/ui/src/composites/wui-icon-box/index.ts

```diff
@@ -0,0 +1,54 @@
+import { html, LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+import { globalStyles } from '../../utils/ThemeUtil'
+import '../../components/wui-icon'
+import styles from './styles'
+import type { Size } from '../../utils/TypesUtil'
+import type { Color } from '../../utils/TypesUtil'
+
+@customElement('wui-icon-box')
+export class WuiIconBox extends LitElement {
+  public static styles = [globalStyles, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @property() public size: Exclude<Size, 'xs' | 'xxs'> = 'md'
+
+  @property() public backgroundColor: Color = 'blue-100'
+
+  @property() public iconColor: Color = 'blue-100'
+
+  // -- render ------------------------------------------------------- //
+  public render() {
+    let iconSize: Size = 'xxs'
+    switch (this.size) {
+      case 'lg':
+        iconSize = 'lg'
+        break
+      case 'md':
+        iconSize = 'xs'
+        break
+      default:
+        iconSize = 'xxs'
+    }
+
+    const sizeClass = `wui-size-${this.size}`
+    const opacityValue = this.size === 'lg' ? '12%' : '16%'
+
+    return html`
+      <div
+        class="${sizeClass}"
+        style="background-color: var(--wui-rgba-transparent);
```

Comment:
Let's avoid inline styles. Do we really need to differentiate % of opacity between different sizes? Perhaps we just use one middleground value for all? i.e. `14%`. Should confirm with design team first. If they absolutely do need different opacity values here, we can just set default bg color in css for `div` and add 2 classes for diff opacities below that with bg color mix.

Reply:
I agree that inline styles are not the best option, but even if we have only 1 opacity value, we still need to create a class for every color. 

```
 .wui-bg-blue-100 {
    background-color: color-mix(in srgb, var(--wui-color-blue-100) 14%, transparent);
  }

.wui-bg-fg-100 {
    background-color: color-mix(in srgb, var(--wui-color-fg-100) 14%, transparent);
  }
```

My preferred solution is to use inline styles only to create a css variable:

```
<div
  class="${classMap(classes)}"
  style="--wui-bg-value: var(--wui-color-${this.backgroundColor});"
>
  <wui-icon color=${this.iconColor} size=${iconSize}><slot></slot></wui-icon>
</div>
```

Then we only have to create 2 classes and don't break the design: 

  ```
.wui-overlay-sm {
    background-color: color-mix(in srgb, var(--wui-bg-value) 12%, transparent);
  }

  .wui-overlay-md {
    background-color: color-mix(in srgb, var(--wui-bg-value) 16%, transparent);
  }
```

Thoughts @0xAsimetriq ?